k9mail/src/main/java/com/fsck/k9/provider/UnreadWidgetProvider.java

```diff
@@ -47,7 +47,7 @@ public static void updateUnreadCount(Context context) {
     }
 
     public static void updateWidget(Context context, AppWidgetManager appWidgetManager,
-            int appWidgetId, String accountUuid) {
+            int appWidgetId, String accountUuid, String folderName) {
 
         RemoteViews remoteViews = new RemoteViews(context.getPackageName(),
```

Comment:
This method is getting longer and longer and it's difficult to see how it could be split.

Maybe we have to accept some small code duplication and split it into:

```
updateWidgetWithSearchAccount()
updateWidgetWithAccount()
updateWidgetWithFolder()
```

Either that or build a structs or something that can be modified by methods and mean we don't have to pass round lots of properties. Maybe:

`class UnreadWidgetProperties {
   unreadCount;
   title;
   clickIntent;
}`

Just some thoughts. Let me know what you think.